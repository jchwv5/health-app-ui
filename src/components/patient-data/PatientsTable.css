.patientsTable {
    margin-top: 40px;
    margin-left: 20%;
}

.wrapper {
    margin-top: 20px;
    text-align: center;
}

h3 {
    margin-left: 25px;
}

table {
    border-collapse: collapse;
    width: 60%;
    margin-left: 10%;
  }
th {
    border: 1px solid black;
}

tr:nth-child(even) {
    background-color: #f1f1f1;
  }

tr:hover {
    background-color: rgb(148, 194, 196);
}

.noHover {
    pointer-events: none;
}

td{ 
    padding-left: 15px;
    border: 1px solid black;
    height: 50px;
 }

.addPatientButton {
    padding: 10px;
}

.deletePatient {
padding-right: 10px;
border: none;
background-color: #ffffff;
}

.errors {
    width: 200px;
    padding-right: 10px;
    border: none;
    background-color: #ffffff;
}   


.deleteHeader {
    border: none;
    width: 40px;
}

.errorDiv {
    color: red;
}

td {
    position: relative;
}

td > .rowLink{
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    text-decoration: none;
    color: inherit;
}