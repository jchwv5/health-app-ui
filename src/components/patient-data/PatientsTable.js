/* eslint-disable react-hooks/exhaustive-deps */
import React, { useEffect, useState } from 'react';
import { Table } from 'react-bootstrap';
import './PatientsTable.css';
import fetchPatients from './DataViewService';

const PatientsTable = () => {
  // eslint-disable-next-line no-unused-vars
  const [patients, setPatients] = useState([]);
  // eslint-disable-next-line no-unused-vars
  const [apiError, setApiError] = useState(false);
  useEffect(() => {
    fetchPatients(setPatients, setApiError);
  }, []);

  // eslint-disable-next-line no-unused-vars
  const PatientMap = ({ data }) => data.map((patient) => (
    <tr className="clickable-row" key={patient.id} a href="patients/details/">
      <td>{patient.firstName}</td>
      <td>{patient.lastName}</td>
      <td>{patient.age}</td>
      <td>{patient.gender}</td>
    </tr>
  ));

  return (
    <div>
      <button type="submit" className="addPatientButton">
        Add a Patient
      </button>
      <Table>
        <thead>
          <tr>
            <th>First Name</th>
            <th>Last Name</th>
            <th>Age</th>
            <th>Gender</th>
          </tr>
        </thead>
        <tbody>
          <PatientMap data={patients} />
        </tbody>
      </Table>
    </div>
  );
};
export default PatientsTable;
