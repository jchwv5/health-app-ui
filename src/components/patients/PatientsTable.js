/* eslint-disable jsx-a11y/anchor-has-content */
/* eslint-disable no-unused-vars */
/* eslint-disable react-hooks/exhaustive-deps */
import React, { useEffect, useState } from 'react';
import DeleteIcon from '@material-ui/icons/Delete';
import './PatientsTable.css';
import fetchPatients from './DataViewService';
import PatientForm from './PatientForm';
import addPatient from './AddPatientService';
import deletePatientById from './DeletePatientService';

const PatientsTable = () => {
  // eslint-disable-next-line no-unused-vars
  const [patients, setPatients] = useState([]);
  // eslint-disable-next-line no-unused-vars
  const [apiError, setApiError] = useState(false);
  const [showAddPatientModal, setShowAddPatientModal] = useState(false);
  const [deleteCount, setDeleteCount] = useState(0);
  const [deleteError, setDeleteError] = useState(0);

  useEffect(() => {
    fetchPatients(setPatients, setApiError);
  }, [deleteCount, showAddPatientModal]);

  const showModal = () => {
    setShowAddPatientModal(true);
  };

  // eslint-disable-next-line no-unused-vars
  const PatientMap = ({ data }) => data.map((patient) => (
    <tr key={patient.id}>
      <td className="deletePatient">
        <button
          type="submit"
          className="deletePatientButton"
          variant="contained"
          onClick={() => {
            if (patient.encounters.length > 0) {
              setDeleteError(patient.id);
            } else if (patient.encounters.length === 0) {
              deletePatientById(setApiError, setDeleteCount, patient.id);
            }
          }}
        >
          <DeleteIcon />
        </button>
      </td>
      <td>
        {patient.firstName}
        <a href={`patients/details/${patient.id}`} className="rowLink" />
      </td>
      <td>
        {patient.lastName}
        <a href={`patients/details/${patient.id}`} className="rowLink" />
      </td>
      <td>
        {patient.age}
        <a href={`patients/details/${patient.id}`} className="rowLink" />
      </td>
      <td>
        {patient.gender}
        <a href={`patients/details/${patient.id}`} className="rowLink" />
      </td>
      <td className="errors noHover">{deleteError === patient.id && <div className="errorDiv">CANNOT DELETE A PATIENT WITH ENCOUNTERS</div>}</td>
    </tr>
  ));

  return (
    <div>
      <h3 className="pateintsHeader">Patients</h3>
      <table className="patientsTable">
        <thead>
          <tr className="noHover">
            <th className="deleteHeader" />
            <th>First Name</th>
            <th>Last Name</th>
            <th>Age</th>
            <th>Gender</th>
          </tr>
        </thead>
        <tbody>
          <PatientMap data={patients} />
        </tbody>
      </table>
      <div className="wrapper">
        {showAddPatientModal && (
        <PatientForm
          onClose={() => setShowAddPatientModal(false)}
          patientApiCall={addPatient}
          patientId={null}
          formHeader="Add a Patient"
          buttonLabel="Add Patient"
          successMsg="Patient Updated Successfully!"
        />
        )}
        <button type="submit" className="addPatientButton" onClick={() => showModal()}>
          Add a Patient
        </button>
      </div>
    </div>
  );
};
export default PatientsTable;
