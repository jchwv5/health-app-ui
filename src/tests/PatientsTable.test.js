import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import PatientsTable from '../components/patients/PatientsTable';

it('renders successfully', () => {
  const div = document.createElement('div');
  ReactDom.render(<PatientsTable />, div);
  ReactDom.unmountComponentAtNode(div);
});

it('App component matches snapshot', () => {
  const component = renderer.create(<PatientsTable />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
