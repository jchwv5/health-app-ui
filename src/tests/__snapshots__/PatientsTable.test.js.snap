// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches snapshot 1`] = `
<div>
  <h3
    className="pateintsHeader"
  >
    Patients
  </h3>
  <table
    className="patientsTable"
  >
    <thead>
      <tr
        className="noHover"
      >
        <th
          className="deleteHeader"
        />
        <th>
          First Name
        </th>
        <th>
          Last Name
        </th>
        <th>
          Age
        </th>
        <th>
          Gender
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
  <div
    className="wrapper"
  >
    <button
      className="addPatientButton"
      onClick={[Function]}
      type="submit"
    >
      Add a Patient
    </button>
  </div>
</div>
`;
